intrastat_product/models/intrastat_product_declaration.py

```diff
@@ -709,7 +709,7 @@ def _gather_invoices(self, notedict):
                     "amount_accessory_cost_company_currency": 0.0,
                     "transaction_id": intrastat_transaction.id,
                     "product_origin_country_id": product_origin_country.id or False,
-                    "region_code": region_code,
+                    "region_code": region_code or region and region.code or False,
```

Comment:
Please avoid this old-stylish spaghetti code:

```suggestion
                    "region_code": region_code or region.code,
```

Reply:
If `region` may be False, then the method `_get_region` should return an empty recordset.